pkg/status/combinedstatus-resolution.go

```diff
@@ -539,14 +581,20 @@ func handleSelect(scName string, scData *statusCollectorData) *v1alpha1.NamedSta
 					Type:   v1alpha1.TypeString,
 					String: &v,
 				}
-			case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
-				numStr := fmt.Sprintf("%d", v)
+			case int, int8, int16, int32, int64:
+				numStr := strconv.Itoa(int(v.(int64)))
+				col = v1alpha1.Value{
+					Type:   v1alpha1.TypeNumber,
+					Number: &numStr,
+				}
+			case uint, uint8, uint16, uint32, uint64:
+				numStr := strconv.FormatInt(int64(v.(uint64)), 10)
```

Comment:
I do not understand why casting to, and formatting as, a signed integer is the right thing to do here. `strconv.FormatUint` is a thing.